src/braket/aws/aws_quantum_task.py

```diff
@@ -331,8 +340,8 @@ async def _wait_for_completion(
                     f" now."
                 )
                 continue
-            current_metadata = self.metadata()
-            task_status = current_metadata["status"]
+            task_status = self._status()
```

Comment:
One thing to consider is checking the cached result (if self.metadata() already exists) and checking whether it's a terminal state; if it is, then we save a call to GetQuantumTask.

Reply:
Good idea